crates/hir-ty/src/display.rs

```diff
@@ -304,6 +304,7 @@ pub struct HirDisplayWrapper<'a, T> {
 pub enum ClosureStyle {
     /// `impl FnX(i32, i32) -> i32`, where `FnX` is the most special trait between `Fn`, `FnMut`, `FnOnce` that the
     /// closure implements. This is the default.
+    // FIXME: Allow rendering non capturing closures as plain function pointers?
```

Comment:
I guess it can become confusing, and the point when a closure becomes a function pointer will be missed if we render them in the same way. Maybe we can use `fn {closure#xx}(args)` similar to `FnDef` types? If rust had a `FnZero` trait for such closures we could use that.

Reply:
Right, that's actually not a good change, since a closure doesn't turn into a function pointer unless coerced to. I'll remove that fixme again